src/extension.ts

```diff
@@ -32,9 +34,15 @@ export function activate(context: ExtensionContext) {
               : false
           )
           .forEach(async (diagnostic) => {
+
+            const markdownString = new MarkdownString(formatDiagnostic(converter.asDiagnostic(diagnostic), prettify));
```

Comment:
Since asDiagnostic isn't documented can you add a short comment about why the conversion from vscode diagnostic to protocol diagnostic is required?